Small bug fixes

joshc-ai21 opened the pull request:
Small bug fixes according to your comments

joshc-ai21 committed:
Update chat notebook

joshc-ai21 committed:
Update ai21.ipynb

joshc-ai21 committed:
June updates to our docs

joshc-ai21 committed:
Merge pull request #1 from joshc-ai21/june-updates

Changes to chat notebook

joshc-ai21 committed:
updates

joshc-ai21 committed:
updates

joshc-ai21 committed:
update

joshc-ai21 committed:
Bug fixes

joshc-ai21 committed:
Bug fixes

eyurtsev committed:
standard-tests[minor]: Add standard tests for cache (#23357)

Add standard tests for cache abstraction

baskargopinath committed:
Update multimodal_prompts.ipynb (#23301)

fixes #23294

---------

Co-authored-by: Chester Curme <chester.curme@gmail.com>

Joffref committed:
community: Fix OVHcloud 401 Unauthorized on embedding. (#23260)

They are now rejecting with code 401 calls from users with expired or
invalid tokens (while before they were being considered anonymous).
Thus, the authorization header has to be removed when there is no token.

Related to: #23178

---------

Signed-off-by: Joffref <mariusjoffre@gmail.com>

baur-krykpayev committed:
community: Add ZenGuard tool (#22959)

** Description**
This is the community integration of ZenGuard AI - the fastest
guardrails for GenAI applications. ZenGuard AI protects against:

- Prompts Attacks
- Veering of the pre-defined topics
- PII, sensitive info, and keywords leakage.
- Toxicity
- Etc.

**Twitter Handle** : @zenguardai

- [x] **Add tests and docs**: If you're adding a new integration, please
include
  1. Added an integration test
  2. Added colab


- [x] **Lint and test**: Run `make format`, `make lint` and `make test`
from the root of the package(s) you've modified.

---------

Co-authored-by: Nuradil <nuradil.maksut@icloud.com>
Co-authored-by: Nuradil <133880216+yaksh0nti@users.noreply.github.com>

jacoblee93 committed:
docs[patch]: Update chatbot tools how-to guide (#23362)

raveharpaz committed:
Add OCI Generative AI new model support  (#22880)

- [x] PR title: 
community: Add OCI Generative AI new model support
 
- [x] PR message:
- Description: adding support for new models offered by OCI Generative
AI services. This is a moderate update of our initial integration PR
16548 and includes a new integration for our chat models under
/langchain_community/chat_models/oci_generative_ai.py
    - Issue: NA
- Dependencies: No new Dependencies, just latest version of our OCI sdk
    - Twitter handle: NA


- [x] Add tests and docs: 
  1. we have updated our unit tests
2. we have updated our documentation including a new ipynb for our new
chat integration


- [x] Lint and test: 
 `make format`, `make lint`, and `make test` run successfully

---------

Co-authored-by: RHARPAZ <RHARPAZ@RHARPAZ-5750.us.oracle.com>
Co-authored-by: Arthur Cheng <arthur.cheng@oracle.com>

ccurme commented on libs/partners/ai21/langchain_ai21/chat_models.py:
```diff
@@ -19,15 +19,17 @@
 
 
-            model = ChatAI21()
+            model = ChatAI21(api_key=<AI21_API_KEY>)
```
this would also be read from environment variables, no?

joshc-ai21 replied:
Indeed yes. But the example doesn't show loading or setting that variable explicitly, like our other code samples do and I wanted to call it out explicitly. We have the odd behavior of allowing api_key to either be set explicitly or via OS environment key "AI21_API_KEY". I've seen a couple ways of doing this:
1. Setting the key explicitly, as here,
2. `export AI21_API_KEY=your-api-key`
3. ```import os...
    import os
    from getpass import getpass
    os.environ["AI21_API_KEY"] = getpass()```

I just wanted to make setting the key explicit, so that users don't fail and get frustrated. WDYT?

joshc-ai21 replied:
Updated a bit more, and added more context for this specific question. I hope it's okay.

ccurme replied:
Looks good!

jacoblee93 committed:
docs[patch]: Adds callout in LLM concept docs, remove deprecated code (#23361)

CC @baskaryan @hwchase17

tomasonjo committed:
Sanitize backticks from neo4j labels and types for import (#23367)

comsa33 committed:
community: fix issue with nested field extraction in MongodbLoader (#22801)

**Description:** 
This PR addresses an issue in the `MongodbLoader` where nested fields
were not being correctly extracted. The loader now correctly handles
nested fields specified in the `field_names` parameter.

**Issue:** 
Fixes an issue where attempting to extract nested fields from MongoDB
documents resulted in `KeyError`.

**Dependencies:** 
No new dependencies are required for this change.

**Twitter handle:** 
(Optional, your Twitter handle if you'd like a mention when the PR is
announced)

### Changes
1. **Field Name Parsing**:
- Added logic to parse nested field names and safely extract their
values from the MongoDB documents.

2. **Projection Construction**:
- Updated the projection dictionary to include nested fields correctly.

3. **Field Extraction**:
- Updated the `aload` method to handle nested field extraction using a
recursive approach to traverse the nested dictionaries.

### Example Usage
Updated usage example to demonstrate how to specify nested fields in the
`field_names` parameter:

```python
loader = MongodbLoader(
    connection_string=MONGO_URI,
    db_name=MONGO_DB,
    collection_name=MONGO_COLLECTION,
    filter_criteria={"data.job.company.industry_name": "IT", "data.job.detail": { "$exists": True }},
    field_names=[
        "data.job.detail.id",
        "data.job.detail.position",
        "data.job.detail.intro",
        "data.job.detail.main_tasks",
        "data.job.detail.requirements",
        "data.job.detail.preferred_points",
        "data.job.detail.benefits",
    ],
)

docs = loader.load()
print(len(docs))
for doc in docs:
    print(doc.page_content)
```
### Testing
Tested with a MongoDB collection containing nested documents to ensure
that the nested fields are correctly extracted and concatenated into a
single page_content string.
### Note
This change ensures backward compatibility for non-nested fields and
improves functionality for nested field extraction.
### Output Sample
```python
print(docs[:3])
```
```shell
# output sample:
[
    Document(
        # Here in this example, page_content is the combined text from the fields below
        # "position", "intro", "main_tasks", "requirements", "preferred_points", "benefits"
        page_content='all combined contents from the requested fields in the document',
        metadata={'database': 'Your Database name', 'collection': 'Your Collection name'}
    ),
    ...
]
```

---------

Co-authored-by: Bagatur <22008038+baskaryan@users.noreply.github.com>
Co-authored-by: Bagatur <baskaryan@gmail.com>

ccurme committed:
mongodb[patch]: fix CI for python 3.12 (#23369)

eyurtsev committed:
standard-tests[minor]: Add standard tests for BaseStore (#23360)

Add standard tests to base store abstraction. These only work on [str,
str] right now. We'll need to check if it's possible to add
encoder/decoders to generalize

eyurtsev committed:
standard-tests[minor]: Add standard read write test suite for vectorstores (#23355)

Add standard read write test suite for vectorstores

userlerueda committed:
partners: add custom options to MongoDBChatMessageHistory (#22944)

**Description:** Adds options for configuring MongoDBChatMessageHistory
(no breaking changes):
- session_id_key: name of the field that stores the session id
- history_key: name of the field that stores the chat history
- create_index: whether to create an index on the session id field
- index_kwargs: additional keyword arguments to pass to the index
creation

**Discussion:**
https://github.com/langchain-ai/langchain/discussions/22918
**Twitter handle:** @userlerueda

---------

Co-authored-by: Jib <Jibzade@gmail.com>
Co-authored-by: Eugene Yurtsev <eugene@langchain.dev>

jacoblee93 committed:
docs[patch]: Fix typo in how to guide for message history (#23364)

smziaurrashid committed:
package: security update urllib3 to @1.26.19 (#23366)

urllib3 version update 1.26.18 to 1.26.19 to address a security
vulnerability.

**Reference:**
https://security.snyk.io/vuln/SNYK-PYTHON-URLLIB3-7267250

eltociear committed:
docs: update sql_query_checking.ipynb (#23345)

creat -> create

yuncliu committed:
community[minor]: Add Ascend NPU optimized Embeddings (#20260)

- **Description:** Add NPU support for embeddings

---------

Co-authored-by: Bagatur <22008038+baskaryan@users.noreply.github.com>
Co-authored-by: Bagatur <baskaryan@gmail.com>

Zizo-Vi committed:
partners: add lint docstrings for azure-dynamic-sessions/together modules (#23303)

Description: add lint docstrings for azure-dynamic-sessions/together
modules
Issue: #23188 @baskaryan

test: ruff check passed.
<img width="782" alt="image"
src="https://github.com/langchain-ai/langchain/assets/76683249/bf11783d-65b3-4e56-a563-255eae89a3e4">

---------

Co-authored-by: gongwn1 <gongwn1@lenovo.com>

mirnawong1 committed:
Replace llm variable with model (#23280)

The code snippet under ‘pdfs_qa’ contains an small incorrect code
example , resulting in users getting errors. This pr replaces ‘llm’
variable with ‘model’ to help user avoid a NameError message.

Resolves #22689


If no one reviews your PR within a few days, please @-mention one of
baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

---------

Co-authored-by: Chester Curme <chester.curme@gmail.com>

rahul-trip committed:
community[minor]: Added classification_location parameter in PebbloSafeLoader. (#22565)

Description: Add classifier_location feature flag. This flag enables
Pebblo to decide the classifier location, local or pebblo-cloud.
Unit Tests: N/A
Documentation: N/A

---------

Signed-off-by: Rahul Tripathi <rauhl.psit.ec@gmail.com>
Co-authored-by: Rahul Tripathi <rauhl.psit.ec@gmail.com>

maang-h committed:
docs: Update QianfanChatEndpoint ChatModel docstring (#23337)

- **Description:** Update QianfanChatEndpoint ChatModel rich docstring
- **Issue:** the issue #22296

Zizo-Vi committed:
community: FlashrankRerank support loading customer client (#23350)

Description: FlashrankRerank Document compressor support loading
customer client
Issue: #23338

Co-authored-by: gongwn1 <gongwn1@lenovo.com>

ccurme committed:
langchain[patch]: update removal target for LLMChain (#23373)

to 1.0

Also improve replacement example in docstring.

joshc-ai21 committed:
Merge branch 'langchain-ai:master' into june-updates

asafgardin committed:
Merge remote-tracking branch 'origin/june-updates' into june-updates

asafgardin committed:
fix: Model name in AI21LLM to J2

asafgardin committed:
fix: Renamed

asafgardin committed:
fix: Chat docs string

asafgardin committed:
docs: Updated AI21LLM doc string

Josephasafg committed:
Merge branch 'master' into june-updates

joshc-ai21 committed:
Update ai21.ipynb

Fixing long lines

joshc-ai21 committed:
Merge branch 'master' into june-updates

joshc-ai21 committed:
Fixing lint issues

joshc-ai21 committed:
Fixing lint issues

Josephasafg committed:
Merge branch 'master' into june-updates

joshc-ai21 committed:
Update ai21.ipynb to fix broken JSON

ccurme committed:
fix typo enviorn -> environ
